cmd/db/per_tenant_migration.go

```diff
@@ -0,0 +1,71 @@
+package db
+
+import (
+	"context"
+	"embed"
+	"fmt"
+
+	migrate "github.com/rubenv/sql-migrate"
+	"github.com/stellar/go/support/log"
+
+	"github.com/stellar/stellar-disbursement-platform-backend/cmd/utils"
+	"github.com/stellar/stellar-disbursement-platform-backend/db"
+	"github.com/stellar/stellar-disbursement-platform-backend/stellar-multitenant/pkg/tenant"
+)
+
+// executeMigrationsPerTenant executes the migrations on the database for all tenants or a specific tenant, according
+// with the direction and count.
+func executeMigrationsPerTenant(ctx context.Context, databaseURL string, opts utils.TenantRoutingOptions, dir migrate.MigrationDirection, count int, migrationFiles embed.FS, tableName db.MigrationTableName) error {
+	if err := opts.ValidateFlags(); err != nil {
+		log.Ctx(ctx).Fatal(err.Error())
+	}
+
+	tenantIDToDNSMap, err := getTenantIDToDSNMapping(ctx, databaseURL)
+	if err != nil {
+		return fmt.Errorf("getting tenants schemas: %w", err)
+	}
+
+	if opts.TenantID != "" {
+		if dsn, ok := tenantIDToDNSMap[opts.TenantID]; ok {
+			tenantIDToDNSMap = map[string]string{opts.TenantID: dsn}
+		} else {
+			log.Ctx(ctx).Fatalf("tenant ID %s does not exist", opts.TenantID)
+		}
```

Comment:
Do we want to log a fatal error here? We can return an error instead. WDYT?

Reply:
Nice catch, thank you! Fixing it in https://github.com/stellar/stellar-disbursement-platform-backend/pull/136/commits/8add269d096cedf1cfb6c336fd0093742c282f52.